boards/FSM/Src/Io/Io_FlowMeters.c

```diff
@@ -0,0 +1,84 @@
+#include "main.h"
+#include "Io_SharedFrequencyOnlyPwmInput.h"
+#include "Io_FlowMeters.h"
+#include <assert.h>
+
+#define PRIMARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_1
+#define SECONDARY_FLOW_METER_CHANNEL HAL_TIM_ACTIVE_CHANNEL_2
+
+struct FlowMeters_Context
+{
+    struct FreqOnlyPwmInput *pwm_input;
+    size_t                   elapsed_tick;
+};
+
+static TIM_HandleTypeDef *       flow_meters_htim;
+static struct FlowMeters_Context primary_flow_meter, secondary_flow_meter;
+
+void Io_FlowMeters_Init(TIM_HandleTypeDef *htim)
+{
+    assert(htim != NULL);
+
+    flow_meters_htim                = htim;
+    primary_flow_meter.elapsed_tick = secondary_flow_meter.elapsed_tick = 0U;
```

Comment:
No point to save one line here imo
```suggestion
    primary_flow_meter.elapsed_tick = 0U;
    secondary_flow_meter.elapsed_tick = 0U;
```